src/adapter/src/catalog.rs

```diff
@@ -5192,6 +5253,86 @@ impl Catalog {
+                                role_name: group_role.name().to_string(),
+                                member_name: member_role.name().to_string(),
+                            },
+                        )));
+                    }
+                    let member_role = state.get_role_mut(&member_id);
+                    member_role.membership.map.insert(role_id, grantor_id);
+                    tx.update_role(member_id, member_role.clone().into())?;
+                    builtin_table_updates
+                        .push(state.pack_role_members_update(role_id, member_id, 1));
+
+                    state.add_to_audit_log(
+                        oracle_write_ts,
+                        session,
+                        tx,
+                        builtin_table_updates,
+                        audit_events,
+                        EventType::Grant,
+                        ObjectType::Role,
+                        EventDetails::GrantRevokeRoleV1(mz_audit_log::GrantRevokeRoleV1 {
+                            role_id: role_id.to_string(),
+                            member_id: member_id.to_string(),
+                            grantor_id: Some(grantor_id.to_string()),
+                        }),
+                    )?;
+                }
+                Op::RevokeRole { role_id, member_id } => {
+                    let member_role = state.get_role(&member_id);
+                    if is_reserved_name(&member_role.name) {
+                        return Err(AdapterError::Catalog(Error::new(
+                            ErrorKind::ReservedRoleName(member_role.name.clone()),
+                        )));
+                    }
+                    let group_role = state.get_role(&role_id);
+                    if is_reserved_name(group_role.name()) {
+                        return Err(AdapterError::Catalog(Error::new(
+                            ErrorKind::ReservedRoleName(group_role.name().to_string()),
+                        )));
+                    }
+                    builtin_table_updates
+                        .push(state.pack_role_members_update(role_id, member_id, -1));
+                    let member_role = state.get_role_mut(&member_id);
+                    member_role.membership.map.remove(&role_id);
+                    tx.update_role(member_id, member_role.clone().into())?;
+
+                    state.add_to_audit_log(
+                        oracle_write_ts,
+                        session,
+                        tx,
+                        builtin_table_updates,
+                        audit_events,
+                        EventType::Revoke,
+                        ObjectType::Role,
+                        EventDetails::GrantRevokeRoleV1(mz_audit_log::GrantRevokeRoleV1 {
+                            role_id: role_id.to_string(),
+                            member_id: member_id.to_string(),
+                            grantor_id: None,
```

Comment:
Light preference to have separate Grant and Revoke EventDetails structs instead of an Option.

Reply:
Updated.